source/frontend/.dependency-cruiser.js

```diff
@@ -0,0 +1,437 @@
+/** @type {import('dependency-cruiser').IConfiguration} */
+module.exports = {
+  forbidden: [
+    /* rules from the 'recommended' preset: */
+    {
+      name: 'no-circular',
+      severity: 'warn',
+      comment:
+        'This dependency is part of a circular relationship. You might want to revise ' +
```

Comment:
given that this pr alters the whole codebase, I recommend to take out this change. That way we can assess the impact of adding this new tool (dependency cruiser) in a separate pr

Reply:
sure thing